File: backend/src/main/kotlin/com/example/backend/model/ApprovableBaseModel.kt
package com.example.backend.model

import javax.persistence.MappedSuperclass

enum class PendingState {
    APPROVED,
    UNAPPROVED,
    TO_EDIT
}

@MappedSuperclass
abstract class ApprovableBaseModel : BaseModel() {
    var pending: PendingState = PendingState.UNAPPROVED
    var reason: String = ""
}

File: frontend-react-ts/src/models/ApprovableBaseModel.ts
import BaseModel from "./BaseModel";
export enum PendingState {
    APPROVED,
    UNAPPROVED
}


export default class ApprovableBaseModel extends BaseModel {
    private reason: String;
    pending: PendingState;

    constructor(id: number, pending: PendingState, reason: String) {
        super(id);
        this.pending = pending;
        this.reason = reason;
    }
}